src/java.naming/share/classes/com/sun/jndi/ldap/Connection.java

```diff
@@ -684,6 +682,43 @@ void cleanup(Control[] reqCtls, boolean notifyParent) {
         }
     }
 
+    // flush and close output stream
+    private void flushCloseOutputStream() {
+        try {
+            outStream.flush();
+        } catch (IOException ioEx) {
+            if (debug)
+                System.err.println("Connection.flushOutputStream: OutputStream flush problem " + ioEx);
+        }
+        try {
+            outStream.close();
+        } catch (IOException ioEx) {
+            if (debug)
+                System.err.println("Connection.closeOutputStream: OutputStream close problem " + ioEx);
+        }
+    }
+
+    // close socket
+    private void closeConnectionSocket() {
```

Comment:
'closeSocketConnection' will be more meaningful.

Reply:
the suggested name was to convey the closing of the  Connection's socket, not a socket connection per se.